lib/Auth/Process/PerunEntitlementExtended.php

```diff
@@ -0,0 +1,148 @@
+    private $entitlementAuthority;
+    private $groupNameAARC;
+    private $adapter;
+
+    public function __construct($config, $reserved)
+    {
+        parent::__construct($config, $reserved);
+        $modulePerunConfiguration = Configuration::getConfig(self::CONFIG_FILE_NAME);
+        assert('is_array($config)');
+
+        $configuration = Configuration::loadFromArray($config);
+
+        $this->outputAttrName = $configuration->getString(self::OUTPUT_ATTR_NAME, 'eduPersonEntitlementExtended');
+        $this->releaseForwardedEntitlement = $configuration->getBoolean(self::RELEASE_FORWARDED_ENTITLEMENT, true);
+        $this->forwardedEduPersonEntitlement = $configuration->getString(
+            self::FORWARDED_EDU_PERSON_ENTITLEMENT,
+            $this->releaseForwardedEntitlement ? Configuration::REQUIRED_OPTION : ''
+        );
+
+        $this->groupNameAARC = $modulePerunConfiguration->getBoolean(self::GROUPNAMEAARC_ATTR, false);
+        $this->entitlementPrefix = $modulePerunConfiguration->getString(
+            self::ENTITLEMENTPREFIX_ATTR,
+            $this->groupNameAARC ? Configuration::REQUIRED_OPTION : ''
+        );
+        $this->entitlementAuthority = $modulePerunConfiguration->getString(
+            self::ENTITLEMENTAUTHORITY_ATTR,
+            $this->groupNameAARC ? Configuration::REQUIRED_OPTION : ''
+        );
+
+        $interface = $configuration->getValueValidate(
+            self::INTERFACE_PROPNAME,
+            [Adapter::RPC, Adapter::LDAP],
+            Adapter::RPC
+        );
+        $this->adapter = Adapter::getInstance($interface);
+    }
+
+    public function process(&$request)
+    {
+        $eduPersonEntitlementExtended = [];
+        $capabilities = [];
+        $forwardedEduPersonEntitlement = [];
+
+        if (isset($request['perun']['groups'])) {
+            $eduPersonEntitlementExtended = $this->getEduPersonEntitlementExtended($request);
+
+            $capabilities = EntitlementUtils::getCapabilities(
+                $request,
+                $this->adapter,
+                $this->entitlementPrefix,
+                $this->entitlementAuthority
+            );
+        } else {
+            Logger::debug(
+                'perun:PerunEntitlementExtended: There are no user groups assigned to facility.' .
+                '=> Skipping getEduPersonEntitlementExtended and getCapabilities'
+            );
+        }
+
+        if ($this->releaseForwardedEntitlement) {
+            $forwardedEduPersonEntitlement = EntitlementUtils::getForwardedEduPersonEntitlement(
+                $request,
+                $this->adapter,
+                $this->forwardedEduPersonEntitlement
+            );
+        }
+
+        $request['Attributes'][$this->outputAttrName] = array_unique(array_merge(
+            $eduPersonEntitlementExtended,
+            $forwardedEduPersonEntitlement,
+            $capabilities
+        ));
+    }
+
+    private function getEduPersonEntitlementExtended(&$request)
```

Comment:
I see three output values for one group:
```
group:NAME
group:UUID
groupAttributes:UUID?...
```
Wasn't it agreed to leave out the NAME version?